### PR TITLE
NI-FAKE: Update metadata to 22.5.0d9, fix _mock_helper

### PR DESCRIPTION
**Pull request by olsl21**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~
~- [ ] I've added tests applicable for this pull request~

### What does this Pull Request accomplish?

- Update NI-FAKE metadata to 22.5.0d9
  - New additions:
    - `custom_struct_typedef` and `custom_struct_nested_typedef` custom types
    - `AltColor` enum
    - `FunctionWithRepeatedCapabilityType` and `GetCustomTypeTypedef`
  - Create custom type implementation files for the new `custom_struct_typedef` and `custom_struct_nested_typedef` custom types
  - Update nifake.mak to copy the custom type implementation files so that the build can succeed
  - Create unit test for the new `get_custom_type_typedef()` function that uses the new custom types
- Fix `_mock_helper.py.mako` codegen logic for functions with custom type parameter to use the actual parameter name instead of hardcoded "cs"

### What testing has been done?

Local build succeeded.

### CODE REVIEW

File: src/nifake/metadata/attributes.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-# This file is generated from NI-FAKE API metadata version 21.0.0d11
+# This file is generated from NI-FAKE API metadata version 22.5.0d9999
```
Comment (olsl21) [inline src/nifake/metadata/attributes.py:2]: I'll update this to an official export once the internal PR gets approved and merged.

**Comment by marcoskirsch** [comment]
What is this PR really doing?
It's not just "updating metadata", it's adding new things to it. This is the "what" and "updating metadata" is the how.
Please improve the description and call the fake driver "NI-FAKE" (similar to NI-DMM or NI-FGEN).

**Comment by marcoskirsch** [comment]
Is it possible to do this in smaller, distinct, single purpose PRs?

First one would be, it seems, to update NI-FAKE metadata to something newer but that doesn't attempt to exercise the new capabilities you are adding to nimi-python (namely the ability for enums to specify what values they convert to when exposed in nimi-python APIs).

File: src/nifake/metadata/attributes.py (hunk 1 above)
Comment (olsl21) [reply src/nifake/metadata/attributes.py]: I will split this PR into smaller PRs and this PR will now updata the NI-FAKE metadata to 22.5.0d9

**Comment by olsl21** [comment]
> Is it possible to do this in smaller, distinct, single purpose PRs?
> 
> First one would be, it seems, to update NI-FAKE metadata to something newer but that doesn't attempt to exercise the new capabilities you are adding to nimi-python (namely the ability for enums to specify what values they convert to when exposed in nimi-python APIs).

I've split it into two PRs (i.e. this PR which updates the metadata to 22.5.0d9 (without the new enum with converter and its related attribute) and #1751 which adds the new enum with converter and its related attribute).

> What is this PR really doing? It's not just "updating metadata", it's adding new things to it. This is the "what" and "updating metadata" is the how. Please improve the description and call the fake driver "NI-FAKE" (similar to NI-DMM or NI-FGEN).

This PR is now updating the metadata to 22.5.0d9 with new custom types, enum and functions, as well as some fixes. I have updated the PR description to list out the changes in the metadata more explicitly.

**Review by marcoskirsch — CHANGES_REQUESTED** [review]
There are two methods being added to `nifake` in this PR but they aren't unit tested.
I think we need a more realistic struct so that it can be properly exercised.

**Comment by olsl21** [comment]
> There are two methods being added to `nifake` in this PR but they aren't unit tested. I think we need a more realistic struct so that it can be properly exercised.

I have completed the implementations for the new custom types and added unit tests for the two new methods in my latest commit.

**Comment by marcoskirsch** [comment]
[CI Error](https://github.com/ni/nimi-python/pull/1750/checks?check_run_id=6917155578)
```
The following code generated files do not match what is commited to Git. Run codegen and include the generated files in the PR.
 M generated/nifake/nifake/unit_tests/test_session.py
```

**Comment by olsl21** [comment]
Sorry for the CI error, I have added in the missing code generated file.

**Review by sbethur — APPROVED** [review]

**Review by marcoskirsch — APPROVED** [review]

File: src/nifake/custom_types/custom_struct_nested_typedef.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import ctypes
+
+import nifake.custom_struct as custom_struct
+import nifake.custom_struct_typedef as custom_struct_typedef
+
+
+# These custom type classes are created as placeholders
```
[the remark this answers is not in the document]
Comment (olsl21) [reply src/nifake/custom_types/custom_struct_nested_typedef.py]: Oh, this comment was actually from before I added the full implementation for these new custom types. I've removed it in my latest commit.

**Pull request merged**